exams/pearson.py

```
@@ -0,0 +1,147 @@
+                tsv_writer.writerow(_map_row(row))
+                valid_rows.append(row)
+            except InvalidTsvRow:
+                log.exception("Invalid tsv row")
+                invalid_rows.append(row)
+
+        return (valid_rows, invalid_rows)
+
+    return _writer
+
+
+def optional_field(field, value):
+    """
+    Returns value if it is not None, otherwise an empty string
+    """
+    value = get_field_mapper(field)(value)
+    return value if value else ''
+
+
+def profile_country_to_alpha3(profile):
+    """
+    Returns the alpha3 code of a profile's country
+    """
+    # Pearson requires ISO-3166 alpha3 codes, but we store as alpha2
+    try:
+        country = pycountry.countries.get(alpha_2=profile.country)
+    except KeyError:
+        raise InvalidProfileDataException()
+    return country.alpha_3
+
+
+ccd_writer = writer([
+    ('ClientCandidateId', 'student_id'),
+    ('FirstName', 'romanized_first_name'),
+    ('LastName', 'romanized_last_name'),
+    ('Email', 'user.email'),
+    ('Address1', 'address1'),
+    ('Address2', partial(optional_field, 'address2')),
+    ('Address3', partial(optional_field, 'address3')),
+    ('City', 'city'),
+    ('State', 'state_or_territory'),
+    ('PostalCode', 'postal_code'),
+    ('Country', profile_country_to_alpha3),
+    ('Phone', 'phone_number'),
+    ('PhoneCountryCode', 'phone_country_code'),
+    ('LastUpdate', lambda profile: format_datetime(profile.updated_on)),
+], field_prefix='profile')
+
+
+def upload_tsv(file_path):
+    """
+    Upload the given TSV files to the remote
+    """
+    cnopts = pysftp.CnOpts()
+    cnopts.hostkeys = None  # ignore knownhosts
+    with pysftp.Connection(
+        host=str(settings.EXAMS_SFTP_HOST),
+        port=int(settings.EXAMS_SFTP_PORT),
+        username=str(settings.EXAMS_SFTP_USERNAME),
+        password=str(settings.EXAMS_SFTP_PASSWORD),
+        cnopts=cnopts,
+    ) as sftp:
+        with sftp.cd(str(settings.EXAMS_SFTP_UPLOAD_DIR)):
```

Why do we need to use `str(...)` here? If `EXAMS_SFTP_UPLOAD_DIR` is not set we should raise an exception here. If there is a sensible default we should define it in `settings.py`